react/features/welcome/components/AbstractWelcomePage.js

```diff
@@ -34,15 +35,32 @@ export class AbstractWelcomePage extends Component {
         /**
          * Save room name into component's local state.
          *
-         * @type {{room: string}}
+         * @type {Object}
+         * @property {string} room - Room name.
+         * @property {string} roomPlaceholder - Room placeholder
+         * that's used as a placeholder for input.
+         * @property {string} generatedRoomname - Automatically generated
+         * room name.
+         * @property {number|null} animateTimeoutId - Identificator for
+         * letter animation timeout.
+         * @property {nubmer|null} updateTimeoutId - Identificator for
+         * updating generated room name.
          */
         this.state = {
-            room: ''
+            room: '',
+            roomPlaceholder: '',
+            generatedRoomname: '',
+            animateTimeoutId: null,
+            updateTimeoutId: null
         };
 
         // Bind event handlers so they are only bound once for every instance.
-        this._onJoinClick = this._onJoinClick.bind(this);
+        const roomnameChanging = this._animateRoomnameChanging.bind(this);
+
+        this._onJoin = this._onJoin.bind(this);
         this._onRoomChange = this._onRoomChange.bind(this);
+        this._updateRoomname = this._updateRoomname.bind(this);
+        this._animateRoomnameChanging = roomnameChanging.bind(this);
```

Comment:
@BeatC, what's the point of the following?

````javascript
const roomnameChanging = this._animateRoomnameChanging.bind(this);
this._animateRoomnameChanging = roomnameChanging.bind(this);
````

Isn't that effectively the following?

````javascript
this._animateRoomnameChanging = this._animateRoomnameChanging.bind(this).bind(this);
````